src/data-workspace/data-entry-cell/data-entry-cell.js

```diff
@@ -0,0 +1,134 @@
+import { IconMore16, colors } from '@dhis2/ui'
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import React, { useState } from 'react'
+import { useField } from 'react-final-form'
+import { useContextSelection } from '../../context-selection/index.js'
+import { useMetadata } from '../metadata-context.js'
+import { getDataSetById } from '../selectors.js'
+import { useDataValueMutation } from '../use-data-value-mutation.js'
+import styles from './data-entry-cell.module.css'
+import { getValidatorByValueType } from './field-validation.js'
+import { useFieldNavigation } from './use-field-navigation.js'
+
+export function DataEntryCell({ dataElement: de, categoryOptionCombo: coc }) {
+    // This field name results in this structure for the form data object:
+    // { [deId]: { [cocId]: value } }
+    const fieldName = `${de.id}.${coc.id}`
+    const validate = getValidatorByValueType(de.valueType)
+    const { input, meta } = useField(fieldName, { validate })
+
+    const [lastSyncedValue, setLastSyncedValue] = useState(meta.initial)
+    const { focusNext, focusPrev } = useFieldNavigation(fieldName)
+
+    const { mutate, isIdle, isLoading, isError } = useDataValueMutation()
+    const [dataEntryContext] = useContextSelection()
+    const { metadata } = useMetadata()
+
+    const syncData = () => {
+        const {
+            dataSetId,
+            orgUnitId,
+            periodId,
+            attributeOptionComboSelection,
+        } = dataEntryContext
+
+        const attributeComboId = getDataSetById(metadata, dataSetId)
+            .categoryCombo.id
+        const isDefaultAttributeCombo =
+            metadata.categoryCombos[attributeComboId].isDefault
+
+        const mutationVars = {
+            de: de.id,
+            co: coc.id,
+            ds: dataSetId,
+            ou: orgUnitId,
+            pe: periodId,
+            value: input.value,
+        }
+        // Add attribute params to mutation if relevant
+        if (!isDefaultAttributeCombo) {
+            // convert ['<catId>:<catOptId>', ...] into 'catOptId1;catOptId2':
+            const attributeOptionIdList = attributeOptionComboSelection
+                .map((aoc) => aoc.split(':')[1])
```

Comment:
Reminder to update this once merged with data-workspace.
`const attributeOptionIdList = Object(attributeOptionComboSelection).values()`